File: frontend/src/types/recruitment.ts
import type { RecommendationLevel } from './analysis';

export interface RecruitmentMetric {
  label: string;
  value: number | string;
  delta?: string;
  tone?: 'blue' | 'green' | 'amber' | 'red' | 'slate';
}

export interface RecruitmentFunnelStage {
  key: string;
  label: string;
  count: number;
  conversion?: number;
}

export interface RecruitmentPipelineColumn {
  key: string;
  title: string;
  description: string;
  count: number;
  candidateNames: string[];
}

export interface CompanyTalentInsight {
  label: string;
  value: string;
  description: string;
  icon: string;
}

export interface JobProfileSuggestion {
  title: string;
  currentSignal: string;
  suggestion: string;
  impact: string;
}

export interface RecruitmentRiskInsight {
  title: string;
  description: string;
  severity: 'low' | 'medium' | 'high';
}

export interface WorkbenchEntry {
  title: string;
  description: string;
  icon: string;
  href: string;
  tag: string;
}

export interface RecruitmentOverview {
  updated_at?: string | null;
  metrics: RecruitmentMetric[];
  funnel: RecruitmentFunnelStage[];
  pipeline: RecruitmentPipelineColumn[];
  companyTalent: CompanyTalentInsight[];
  jobProfileSuggestions: JobProfileSuggestion[];
  riskInsights: RecruitmentRiskInsight[];
  entries: WorkbenchEntry[];
}

export interface TalentPoolCandidate {
  id: string;
  analysisId: string;
  resumeId: string;
  candidateName: string;
  jobId: string;
  jobTitle: string;
  score: number;
  recommendation: RecommendationLevel;
  recommendationLabel: string;
  skills: string[];
  riskFlags: string[];
  summary: string;
  analyzedAt?: string | null;
}

export interface TalentPoolResponse {
  items: TalentPoolCandidate[];
  total: number;
  generatedAt?: string | null;
}

export interface CandidateReportSection {
  title: string;
  items: string[];
}

export interface CandidateReport {
  id: string;
  candidateName: string;
  candidateEmail?: string | null;
  jobTitle: string;
  score: number;
  recommendation: RecommendationLevel;
  recommendationLabel: string;
  recommendationReasons: string[];
  riskPoints: string[];
  interviewQuestions: string[];
  outreachScripts: string[];
  skills: string[];
  generatedAt?: string | null;
}

File: backend/app/schemas/recruitment.py
from datetime import datetime
import uuid

from pydantic import BaseModel


class RecruitmentMetric(BaseModel):
    label: str
    value: int | float | str
    delta: str | None = None
    tone: str = "blue"


class RecruitmentFunnelStage(BaseModel):
    key: str
    label: str
    count: int
    conversion: float | None = None


class RecruitmentPipelineColumn(BaseModel):
    key: str
    title: str
    description: str
    count: int
    candidateNames: list[str]


class CompanyTalentInsight(BaseModel):
    label: str
    value: str
    description: str
    icon: str


class JobProfileSuggestion(BaseModel):
    title: str
    currentSignal: str
    suggestion: str
    impact: str


class RecruitmentRiskInsight(BaseModel):
    title: str
    description: str
    severity: str


class WorkbenchEntry(BaseModel):
    title: str
    description: str
    icon: str
    href: str
    tag: str


class RecruitmentOverview(BaseModel):
    updated_at: datetime
    metrics: list[RecruitmentMetric]
    funnel: list[RecruitmentFunnelStage]
    pipeline: list[RecruitmentPipelineColumn]
    companyTalent: list[CompanyTalentInsight]
    jobProfileSuggestions: list[JobProfileSuggestion]
    riskInsights: list[RecruitmentRiskInsight]
    entries: list[WorkbenchEntry]


class TalentPoolCandidate(BaseModel):
    id: str
    analysisId: uuid.UUID
    resumeId: uuid.UUID
    candidateName: str
    jobId: uuid.UUID
    jobTitle: str
    score: float
    recommendation: str
    recommendationLabel: str
    skills: list[str]
    riskFlags: list[str]
    summary: str
    analyzedAt: datetime | None = None


class TalentPoolResponse(BaseModel):
    items: list[TalentPoolCandidate]
    total: int
    generatedAt: datetime


class JobInsightResponse(BaseModel):
    jobId: uuid.UUID
    jobTitle: str
    totalResumes: int
    analyzed: int
    avgScore: float
    recommendedCount: int
    bottlenecks: list[str]
    suggestions: list[str]


class CandidateReport(BaseModel):
    id: uuid.UUID
    candidateName: str
    candidateEmail: str | None = None
    jobTitle: str
    score: float
    recommendation: str
    recommendationLabel: str
    recommendationReasons: list[str]
    riskPoints: list[str]
    interviewQuestions: list[str]
    outreachScripts: list[str]
    skills: list[str]
    generatedAt: datetime | None = None
